src/core/String.cpp

```diff
@@ -555,4 +555,35 @@ signed char String::naturalnocasecmp_to(String p_str) const {
 	return godot::api->godot_string_naturalnocasecmp_to(&_godot_string, &p_str._godot_string);
 }
 
+String String::dedent() const {
+	String new_string;
+	new_string._godot_string = godot::core_1_1_api->godot_string_dedent(&_godot_string);
+	return new_string;
+}
+
+PoolStringArray String::rsplit(const String& divisor, const bool allow_empty,
+	const int maxsplit) const
+{
+	godot_array arr = godot::core_1_1_api->godot_string_rsplit(&_godot_string, &divisor._godot_string, allow_empty, maxsplit);
```

Comment:
There is a problem here, `godot_string_rsplit()` returns a `godot_pool_string_array` and it should return a `godot_array`. This requires a change in the C headers in the engine.

Reply:
It's converted to PoolStringArray in the engine [modules/gdnative/gdnative/string.cpp#L1327](https://github.com/godotengine/godot/blob/019760d611b6a06ce8d65e7ae23c1546c14466d6/modules/gdnative/gdnative/string.cpp#L1327).

Reply:
I'll send a PR fixing it.

Reply:
I've been checking the sources of the headers and I'm a bit confused.

You can find functions like [godot_string_md5_buffer](https://github.com/godotengine/godot/blob/019760d611b6a06ce8d65e7ae23c1546c14466d6/modules/gdnative/gdnative/string.cpp#L1014) defined with `godot_pool_byte_array` as the return. In GDScript it returns a [PoolByteArray](https://docs.godotengine.org/en/latest/classes/class_string.html?highlight=bigrams#class-string-md5-buffer).

But for the same class you find functions like [godot_string_bigrams](https://github.com/godotengine/godot/blob/019760d611b6a06ce8d65e7ae23c1546c14466d6/modules/gdnative/gdnative/string.cpp#L160) returning a `godot_array` while in gds it returns a [PoolStringArray](https://docs.godotengine.org/en/latest/classes/class_string.html?highlight=bigrams#class-string-bigrams).

Which should be used? everything as `godot_array` in the GDNative headers of use the correct PoolArrray?

Reply:
I think it's an error that those functions return `Array`s in GDScript. If you split a String on every occurrence of a character, what do you get back? An array that can hold arbitrary data or an array of strings?

GDNative is supposed to be as statically typed as possible, so whenever we observed that a dynamically typed array *is* (and should) actually be typed, we corrected that in the bindings.